src/lib/pickles_types/hlist.mli

```diff
@@ -0,0 +1,918 @@
+open Poly_types
```

Comment:
btw I'm a big fan of starting documentation with a few examples on how to use a library, which is usually why people are looking at the documentation in the first place